src/main/java/frc/trigon/robot/subsystems/ledstrip/LEDStrip.java

```diff
@@ -1,189 +1,97 @@
+            LEDStripConstants.LED_BUFFER.setHSV(led + indexOffset, hue, 255, 128);
+            LED.setData(LEDStripConstants.LED_BUFFER);
+        }
+        rainbowFirstPixelHue += 3;
+        rainbowFirstPixelHue %= 180;
     }
 
-    void animateSingleFade(Color color, double speed) {
-        CANDLE.animate(
-                new SingleFadeAnimation(
-                        color.getRed(),
-                        color.getGreen(),
-                        color.getBlue(),
-                        0,
-                        speed,
-                        this.numberOfLEDs,
-                        this.offset
-                ),
-                animationSlot
-        );
+    void threeSectionColor(Color firstSectionColor, Color secondSectionColor, Color thirdSectionColor) {
+        final int ledsPerSection = (int) Math.floor(numberOfLEDs / 3.0);
+        setThreeSectionColor(ledsPerSection, firstSectionColor, secondSectionColor, thirdSectionColor);
     }
 
-    void animateTwinkle(Color color, double speed, TwinkleAnimation.TwinklePercent divider) {
-        CANDLE.animate(
-                new TwinkleAnimation(
-                        color.getRed(),
-                        color.getGreen(),
-                        color.getBlue(),
-                        0,
-                        speed,
-                        this.numberOfLEDs,
-                        divider,
-                        this.offset
-                ),
-                animationSlot
-        );
+    private void setThreeSectionColor(int ledsPerSection, Color firstSectionColor, Color secondSectionColor, Color thirdSectionColor) {
+        for (int i = 0; i < ledsPerSection; i++)
+            setLEDColor(inverted ? thirdSectionColor : firstSectionColor, i);
+        for (int i = ledsPerSection; i < 2 * ledsPerSection; i++)
+            setLEDColor(secondSectionColor, i);
+        for (int i = 2 * ledsPerSection; i <= numberOfLEDs; i++)
+            setLEDColor(inverted ? firstSectionColor : thirdSectionColor, i);
     }
 
-    void animateStrobe(Color color, double speed) {
-        CANDLE.animate(
-                new StrobeAnimation(
-                        color.getRed(),
-                        color.getGreen(),
-                        color.getBlue(),
-                        0,
-                        speed,
-                        this.numberOfLEDs,
-                        this.offset
-                ),
-                animationSlot
-        );
+    private void setLEDColor(Color color, int index) {
+        LEDStripConstants.LED_BUFFER.setLED(index + indexOffset, color);
+        LED.setData(LEDStripConstants.LED_BUFFER);
     }
 
-    void clearAnimation() {
-        CANDLE.clearAnimation(animationSlot);
+    private void setAllStripColors(Color color) {
+        for (int index = 0; index <= numberOfLEDs; index++) {
+            setLEDColor(color, index);
+        }
```

Comment:
No need for this redundant function

Reply:
I think it looks cleaner this way

Reply:
You can say clearing the leds is like setting a static color of black, but another function is confusing